articles/outer_joins.rst

```diff
@@ -0,0 +1,247 @@
+How Distributed Outer Joins on PostgreSQL with Citus Work
+#########################################################
+
+SQL is a very powerful language for analyzing and reporting against
+data. At the core of SQL is the idea of joins and how you combine
+various tables together. One such type of join: outer joins are useful
+when we need to retain rows, even if it has no match on the other side.
+
+And while the most common type of join, inner join, against tables A and
+B would bring only the tuples that have a match for both A and B, outer
+joins give us the ability to bring together from say all of table A even
+if they don’t have a corresponding match in table B. For example, let's
+say you keep customers in one table and purchases in another table. When
+you want to see all purchases of customers, you may want to see all
+customers in the result even if they did not do any purchases yet. Then,
+you need an outer join. Within this post we’ll analyze a bit on what
+outer joins are, and then how we support them in a distributed fashion
+on Citus.
+
+Let’s say we have two tables, customer and purchase:
+
+.. code:: sql
+
+    customer table:
+     customer_id |      name
+    -------------+-----------------
+               1 | Corra Ignacio
+               3 | Warren Brooklyn
+               2 | Jalda Francis
+
+    purchase table:
+     purchase_id | customer_id | category |           comment
+    -------------+-------------+----------+------------------------------
+            1000 |           1 | books    | Nice to Have!
+            1001 |           1 | chairs   | Comfortable
+            1002 |           2 | books    | Good Read, cheap price
+            1003 |          -1 | hardware | Not very cheap
+            1004 |          -1 | laptops  | Good laptop but expensive...
+
+The following queries and results help clarifying the inner and outer
+join behaviors:
+
+.. code:: sql
+
+     SELECT customer.name, purchase.comment
+     FROM customer JOIN purchase ON customer.customer_id = purchase.customer_id
+     ORDER BY purchase.comment;
+
+          name      |        comment
+     ---------------+------------------------
+      Corra Ignacio | Comfortable
+      Jalda Francis | Good Read, cheap price
+      Corra Ignacio | Nice to Have!
+
+.. image:: ../images/articles-join-inner.png
```

Comment:
The images in this Article aren't aligned (as they were in the blog post).

This was a bit confusing to me. Is there a way to align them with the sql code blocks?

Reply:
(Done)